.github/workflows/solidity-wrappers.yml

```diff
@@ -0,0 +1,89 @@
+name: Solidity Wrappers
+# This is its own workflow file rather than being merged into "solidity.yml" to avoid over complicating the conditionals
+# used for job execution. The jobs in "solidity.yml" are configured around push events, whereas
+# we only want to generate gethwrappers during pull requests.
+on: 
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  changes:
+    name: Detect changes
+    runs-on: ubuntu-latest
+    outputs:
+      changes: ${{ steps.ch.outputs.changes }}
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      - name: Detect changes
+        id: ch
+        uses: ./.github/actions/detect-solidity-file-changes
+
+  # On a pull request event, make updates to gethwrappers if there are changes.
+  update-wrappers:
+    needs: [changes]
+    if: needs.changes.outputs.changes == 'true' 
+    name: Update Wrappers
+    permissions:
+      actions: read
+      id-token: write
+      contents: read
+    runs-on: ubuntu22.04-8cores-32GB
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+
+      - name: Setup Go
+        uses: ./.github/actions/setup-go
+
+      - name: Setup NodeJS
+        uses: ./.github/actions/setup-nodejs
+        with:
+          prod: "true"
+
+      - name: Run native compile and generate wrappers
+        run: make wrappers-all
+        working-directory: ./contracts
+
+      - name: Assume role capable of dispatching action
```

Comment:
should we just use this composite action instead?
https://github.com/smartcontractkit/.github/blob/main/actions/setup-github-token/action.yml